.github/workflows/deploy.yml

```diff
@@ -39,10 +39,19 @@ jobs:
       if: github.ref == 'refs/heads/beta'
       run: echo "BETA=true" >> $GITHUB_ENV
 
-    - name: Use scripts and templates from master branch
+    - name: Use deploy script from master branch
       run: |
         git fetch --no-tags --prune --depth=1 origin master
-        git checkout origin/master -- .github/deploy.sh util/gh-pages/ util/*.py
+        git checkout origin/master -- .github/deploy.sh
+
+    - name: cargo update
+      run: cargo update
```

Comment:
I'm not sure if this is actually needed. The `Cargo.lock` file is not part of our repository and also not cached based on the documentation. This will not hurt but might be unnecessary.

Reply:
I think it is necessary. The cache action will use a hash of the `Cargo.lock` file in the cache key. If we wouldn't have the `Cargo.lock` file the deploy script would build Clippy with other deps than the other workflows, that build Clippy from scratch, which might break the deploy build. I rather have more cache misses, because the `lock` file changes slightly than having to deal with breakages because of cached deps.

Reply:
The selected caching implementation can also just use the `Cargo.toml` at least according to the documentation. 

> a hash of the Cargo.lock / Cargo.toml files (if present). 
> ~(Source: https://github.com/Swatinem/rust-cache#cache-details)

It might be good to add the `Cargo.locked`to the project, but not necessarily required. If we do, we should also think about how the locked file will be updated in the future. This `cargo update` call doesn't commit the file from my understanding.

Reply:
Yes, it also uses the `Cargo.toml`. But if you have for example `dep_xyz = "0.6"` in your `Cargo.toml`, everything from `0.6.0` up to `0.6.∞` is a valid version to use. If `0.6.2` is currently cached, `cargo` will just use that instead of downloading a new version. But since the main Clippy workflows  build Clippy from scratch they always use the latest versions of the deps. So it could happen, that Clippy in `clippy_bors.yml` is build with `dep_xyz = "0.6.3"`, but the deploy workflow still uses `0.6.2`, since it is cached. This can lead to build failures. This is avoided by running `cargo update`, since it produces the `Cargo.lock` file. I just noticed, that `cargo generate-lockfile` does this also and will switch to that.

Clippy can't have a `Cargo.lock` file since it is part of the `rust-lang/rust` repo and would cause all kinds of problems if it had one.

Reply:
Ahh, I had a misconception about how this part would work. Thank you for clarifying :upside_down_face: